src/main/java/bhs/devilbotz/subsystems/DriveTrain.java

```diff
@@ -75,6 +80,75 @@ public class DriveTrain extends SubsystemBase {
+  private static final TalonSRXSimCollection rightMasterSim = rightMaster.getSimCollection();
+
+  /**
+   * The simulation model of our drivetrain
+   *
+   * @see <a
+   *     href="https://docs.wpilib.org/en/stable/docs/software/wpilib-tools/robot-simulation/drivesim-tutorial/drivetrain-model.html">Creating
+   *     a Drivetrain Model</a>
+   * @see <a
+   *     href="https://github.com/CrossTheRoadElec/Phoenix-Examples-Languages/blob/ccbc278d944dae78c73b342003e65138934a1112/Java%20General/DifferentialDrive_Simulation/src/main/java/frc/robot/Robot.java#L6">CTRE
+   *     DifferentialDrive Simulation Example</a>
+   * @since 1/31/2023
+   */
+  private DifferentialDrivetrainSim differentialDriveSim =
+      new DifferentialDrivetrainSim(
+          // Create a linear system from our identification gains.
+          Constants.DriveConstants.DRIVE_PLANT,
+          Constants.DriveConstants.MOTOR_CONFIGURATION,
+          Constants.DriveConstants.MOTOR_GEAR_RATIO,
+          Constants.DriveConstants.TRACK_WIDTH,
+          Constants.DriveConstants.WHEEL_RADIUS,
+
+          // The standard deviations for measurement noise:
+          // x and y:          0.001 m
+          // heading:          0.001 rad
+          // l and r velocity: 0.1   m/s
+          // l and r position: 0.005 m
+          VecBuilder.fill(0.001, 0.001, 0.001, 0.1, 0.1, 0.005, 0.005));
+
+  /**
+   * Helper function to convert position (in meters) to Talon SRX encoder native units. Used for
+   * Simulation.
+   *
+   * @param positionMeters The robot's current position (in meters)
+   * @return The robot's current position in native units (sensorCount)
+   * @see com.ctre.phoenix.motorcontrol.TalonSRXSimCollection#setQuadratureRawPosition(int)
+   * @see <a
+   *     href="https://github.com/crosstheroadelec/Phoenix-Examples-Languages/blob/ccbc278d944dae78c73b342003e65138934a1112/Java%20General/DifferentialDrive_Simulation/src/main/java/frc/robot/Robot.java#L208">CTRE
+   *     Sample Code</a>
+   * @since 1/31/2023
+   */
+  private int distanceToNativeUnits(double positionMeters) {
+    double wheelRotations = positionMeters / (2 * Math.PI * DriveConstants.WHEEL_RADIUS);
+    double motorRotations = wheelRotations * DriveConstants.ENCODER_GEAR_RATIO;
+    int sensorCounts = (int) (motorRotations * DriveConstants.ENCODER_RESOLUTION);
+    return sensorCounts;
+  }
+
+  /**
+   * Helper function to convert velocity to Talon SRX encoder native units. Used for Simulation.
+   *
+   * @param velocityMetersPerSecond The robot's current velocity (in meter/second)
+   * @see com.ctre.phoenix.motorcontrol.TalonSRXSimCollection#setQuadratureVelocity(int)
+   * @see <a
+   *     href="https://github.com/crosstheroadelec/Phoenix-Examples-Languages/blob/ccbc278d944dae78c73b342003e65138934a1112/Java%20General/DifferentialDrive_Simulation/src/main/java/frc/robot/Robot.java#L215">CTRE
+   *     Sample Code</a>
+   * @since 1/31/2023
```

Comment:
make sure to put a @returns parameter in methods that return something.

Reply:
Ah. I missed that one.  Tnx for pointing it out.